runtime/common/src/migrations.rs

```diff
@@ -59,6 +63,68 @@ where
 	}
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct OldRoundInfo<BlockNumber> {
+	pub current: RoundIndex,
+	pub first: BlockNumber,
+	pub length: u32,
+}
+
+pub struct UpdateFirstRoundRelayBlockNumber<T>(pub PhantomData<T>);
+impl<T> Migration for UpdateFirstRoundRelayBlockNumber<T>
+where
+	T: pallet_parachain_staking::Config,
+	T: cumulus_pallet_parachain_system::Config,
+	T: frame_system::Config,
+	u32: From<<<<T as frame_system::Config>::Block as Block>::Header as Header>::Number>,
+{
+	fn friendly_name(&self) -> &str {
+		"MM_UpdateFirstRoundRelayBlockNumber"
+	}
+
+	fn migrate(&self, _available_weight: Weight) -> Weight {
+		let _ = Round::<T>::translate::<OldRoundInfo<BlockNumberFor<T>>, _>(|v1| {
+			Some(RoundInfo {
+				current: v1
+					.expect("old current round value must be present!")
+					.current,
+				// TODO: re-check
+				// a number close to the relay slot
+				first: 283_960_000u64,
```

Comment:
Actually, we have a way to get the relay slot in the migration, because the migration is executed before the on_initialize hook, sop the storage item `ValidationData` is still populated from the previous block

Reply:
Yes I thought about that, but the problem is that `ValidationData` storage doesn't store the relay slot directly. The relay slot can be known from the `RelayChainStateProof` type (as it's done inside `set_validation_data`). I think only at that point you'd be able to access the keys present in the relay state, but I'll continue digging into it to see if I find a way to fetch it.

Reply:
Ok I see, so I have an idea to solve that:

1. Add attribute `#[pallet::storage_version(1)]` on `Pallet` type
2. Add a Consensus Hook wrapper that read the pallet version and run the migration if `StorageVersion::get::<Pallet<T>>() == 0`
3. In `Staking::on_initialize`, wrap all the body code inside a if `StorageVersion::get::<Pallet<T>>() > 0`

Reply:
Just for visibility, at the end we will use `pallet-async-backing` to fetch the `relay-slot` inserted in the previous `para-block` to be able to perform the checks inside the migration and test it as usual with try-runtime.